target/linux/realtek/files-6.12/drivers/net/dsa/rtl83xx/rtl931x.c

```diff
@@ -697,33 +696,38 @@ static void rtl931x_fill_l2_row(u32 r[], struct rtl838x_l2_entry *e)
 		return;
 	}
 
-	r[2] = BIT(31);	/* Set valid bit */
+	r[0] = BIT(31); /* Set valid bit */
 
-	r[0] = ((u32)e->mac[0]) << 24 |
-	       ((u32)e->mac[1]) << 16 |
-	       ((u32)e->mac[2]) << 8 |
-	       ((u32)e->mac[3]);
-	r[1] = ((u32)e->mac[4]) << 24 |
-	       ((u32)e->mac[5]) << 16;
+	r[0] |= ((u32)e->mac[0]) << 8 |
+	       ((u32)e->mac[1]);
+	r[1] = ((u32)e->mac[2]) << 24 |
+	       ((u32)e->mac[3]) << 16 |
+		   ((u32)e->mac[4]) << 8 |
+		   ((u32)e->mac[5]);
 
-	r[2] |= e->next_hop ? BIT(12) : 0;
+	r[0] |= e->is_open_flow ? BIT(30) : 0;
+	r[0] |= e->is_pe_forward ? BIT(28) : 0;
+	r[2] |= e->next_hop ? BIT(30) : 0;
```

Comment:
Here you add something to `r[2]` but never set it to a known value. We must consider the content of `r[2]` as being random